### PR TITLE
Avoid AppVeyor build failures due to problem uploading JUnit XML

### LINKED ISSUE #18994: Flaky build: AppVeyor build sometimes fails due to problems uploading JUnit XML
In our AppVeyor builds, we sometimes see the build fail when it has trouble uploading the JUnit XML files. For example...

```
...
Finished in 710.703 seconds
2089 tests, 11649 assertions, 0 failures, 8 skipped
Command exited with code 1
$wc = New-Object 'System.Net.WebClient'
$endpoint = "https://ci.appveyor.com/api/testresults/junit/$($env:APPVEYOR_JOB_ID)"
Write-Output "Searching for JUnit XML output beneath $($env:TEST_JUNIT_XML_ROOT)"
Get-ChildItem -Path $env:TEST_JUNIT_XML_ROOT -Recurse -File -Name -Include "*.xml" | ForEach-Object {
  $full = "$($env:TEST_JUNIT_XML_ROOT)\$($_)"
  Write-Output "Uploading JUnit XML file $($full)"
  $wc.UploadFile($endpoint, $full)
}
Searching for JUnit XML output beneath c:\projects\junit-test-results
Uploading JUnit XML file c:\projects\junit-test-results\core-main-process.xml
Uploading JUnit XML file c:\projects\junit-test-results\core-render-process.xml
Exception calling "UploadFile" with "2" argument(s): "The remote server returned an error: (500) Internal Server Error."
At line:7 char:3
+   $wc.UploadFile($endpoint, $full)
+   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : WebException
```

Rebuilding can sometimes resolve the failure.

I wonder if we could enhance the build script to retry the upload if it fails at first. Alternatively, perhaps we could consider catching this error and logging it, but not allowing it to cause the overall build to fail.

---

**Affected CI provider(s)**: AppVeyor

**Affected platform(s)**: Windows

**Example commit experiencing flakiness**: https://github.com/atom/atom/commit/b6bf3a5c5d2d74c998cb953131d516e854893288

**Example failing build at that commit**: https://ci.appveyor.com/project/Atom/atom/builds/23020313/job/aeqv4kml7rhony8o

**Example passing build at that commit**: https://ci.appveyor.com/project/Atom/atom/builds/23023754

**Example log from failing build**: https://github.com/atom/atom/files/2968124/build.log

### PR DESCRIPTION
**Pull request by jasonrudolph**
Closes #18994

### Description of the Change
As seen in #18994, our AppVeyor builds periodically fail due to problems uploading the JUnit XML output. Since we already have JUnit XML output for all of our builds on Azure DevOps, and since we want to retire AppVeyor and move to Azure DevOps, it seems pragmatic to resolve this issue by removing the JUnit XML processing from the AppVeyor build. This pull request currently takes this approach to resolving #18994.

### Alternate Designs
We could enhance the [uploading logic](https://github.com/atom/atom/blob/27ff10409f467f6b95d322da12de514cab3c9db5/appveyor.yml#L101-L108) to gracefully ignore errors that occur during upload and/or retry the upload when it fails. However, since we're hoping to migrate off of AppVeyor, this doesn't feel like a particularly productive endeavor.

### Possible Drawbacks
When a test fails on AppVeyor, developers will need to look for the test failure in the build log ([screenshot](https://user-images.githubusercontent.com/2988/56991382-7f1d3400-6b65-11e9-8985-f6e40e2e8770.png)) instead of finding it on the "Tests" tab ([screenshot](https://user-images.githubusercontent.com/2988/56991383-7f1d3400-6b65-11e9-9c9f-bb15ff9d6e13.png)).

### Verification Process
- [x] Verify that Azure DevOps has offers the JUnit XML goodies for all platforms -- https://github.com/atom/atom/pull/19243#issuecomment-488107339
- [x] Ensure that the AppVeyor build still passes and that you can still see all the build output in the AppVeyor log

### CODE REVIEW

**Comment by jasonrudolph** [comment]
> * [x]  Verify that Azure DevOps has offers the JUnit XML goodies for all platforms

To verify this, I pushed up a temporary branch with some platform-specific tests that would intentionally fail:

https://github.com/atom/atom/blob/e7d77719049650e4f65061cfb98ed83c605e59ff/spec/main-process/a-poc.test.js#L2-L14

I then [ran those tests on Azure DevOps](https://github.visualstudio.com/Atom/_build/results?buildId=39249&view=ms.vss-test-web.build-test-results-tab), and verified that the "Tests" tab shows each of these tests:

Screen Shot 2019-04-30 at 4 17 12 PM

In other words, any test that was failing on Windows on AppVeyor should also be visible in the Windows output on Azure DevOps, so I don't think we're losing any meaningful functionality by removing the JUnit XML support on AppVeyor.

**Comment by jasonrudolph** [comment]
@smashwilson: Given your work in this area in https://github.com/atom/atom/pull/15124, would you mind reviewing this change?

**Review by smashwilson — APPROVED** [review]
👍 Sure, seems reasonable.

Redundancy with Azure isn't everything -- we _have_ seen tests that fail here and not on Azure and vice versa, for whatever reason. (Isn't CI fun?) But I think we can deal with a bit of scrolling until we can move fully off of AppVeyor.
